### ISSUE TITLE
Test our documentation for production environments setups

### ISSUE THREAD

**Issue by Angamanga** [ushahidi/platform#3663]
# Intro 📖 
We are continuously working with our documentation to provide a good and smooth experience when setting up the platform. In this ticket, we are asking for help to improve our documentation for installation for production environments.

# Task ✏️ 
- Go to https://docs.ushahidi.com/platform-developer-documentation/v/set-up-the-developer-environment-for-ushahidi/set-up-your-developer-environment/setting-up-the-api-with-docker
- Follow the instructions and try to install the platform
- Document and write comments about the installation-process as comments below, it could be for example: 
  - Confusing instructions
  - Errors you get that is not mentioned in the documentation
  - Do you get stuck somewhere and don't manage to get past a point in the documentation? Where is that?
  - Do you have to do things that is not mentioned in the documentation to make it work? Where and what did you do?
  - Anything else that comes to mind, suggestions, comments etc when reading through the docs.

Thank you for your time and help 🎉 🙏 🙇‍♀

**Comment by ford-prefect** [comment]
The documentation does not mention `API_KEY` in `platform/.env` which is needed for the oauth keys to be generated. I don't know if that would have made the `migrate` script work, but I figured this out later and manually generated the keys.

**Comment by ford-prefect** [comment]
Also the default deployment produces an 'Access Denied"' when you open an URL, which is misleading, because you can skip login and navigate to the pages you should have access to without a login. Still not sure how to fix this one.

**Comment by muchirijane** [comment]
hey @Angamanga, If the issue is still open, could please I work on it.

**Comment by Angamanga** [comment]
@muchirijane I assigned you to this one so start with that: https://github.com/ushahidi/platform/issues/3662

**Comment by amycruz97** [comment]
can i take on this issue

**Comment by hyounes4560** [comment]
@rowasc @Angamanga Could you please allow me to work on this task? Thanks a lot

**Comment by Angamanga** [comment]
@amycruz97 @hyounes4560 Yep, please go ahead. This issue is open for multiple contributors. Please add your findings as comments to this issue :)

**Comment by Naimaoye** [comment]
can I also work on this issue? @Angamanga

**Comment by ulomau** [comment]
@angamanga please can i join and work on this issue?

**Comment by ulomau** [comment]
Hello @Angamanga, i am an Outreachy intern looking for issues to work on. Please can i also join and work on this issue

**Comment by dariatsvetkova** [comment]
Hi @Angamanga, I would also like to work on this issue. And I have a suggestion right away:

Current documentation [Development environment setup with Vagrant](https://docs.ushahidi.com/platform-developer-documentation/development-and-code/setup_alternatives/vagrant-setup) mentions different versions of PHP depending on which version of the platform is used. However, later in the manual, there is a command `sudo update-alternatives --set php /usr/bin/php7.1`

I've noticed that a couple of people on gitter ran into an error installing composer afterwards since it requires PHP version >=7.2 < 7.4. So maybe it would make sense to put `sudo update-alternatives --set php /usr/bin/php7.2` in the manual, or add a comment about different PHP versions after it?

**Comment by treasurechristain** [comment]
@Angamanga I'm Amadikwa Joy N. An outreachy Applicant. Can I work on this issue please.

**Comment by Angamanga** [comment]
@treasurechristain Yes :) Please go ahead!

**Comment by treasurechristain** [comment]
@Angamanga this link https://docs.ushahidi.com/platform-developer-documentation/getting-started/setup_alternatives/installing-for-production-environments is taking me to this environment.
![misleadinglink](https://user-images.githubusercontent.com/20610726/96052797-284d0c80-0e76-11eb-9b37-4ae235e9a47a.PNG)

I guess this is the right link to the issue
https://docs.ushahidi.com/platform-developer-documentation/development-and-code/setup_alternatives/installing-for-production-environments

**Comment by Angamanga** [comment]
@dariatsvetkova @treasurechristain Thank you for your suggestions! It is very useful, we'll review shortly 🙏

**Comment by CeciliaHinga** [comment]
@Angamanga hacktoberfest?

**Comment by Bilalshaikh15** [comment]
@Angamanga the setting up the project doc link is incorrect. I think the correct link would be this 
https://docs.ushahidi.com/platform-developer-documentation/v/set-up-the-developer-environment-for-ushahidi/

**Comment by Codegirlemem** [comment]
I have followed the link above and tried setting up the API with docker and I got stuck at the "Install Make command for the OS you use (parses Makefile)"step. I cannot move past it. Unlike the installation of Docker and repo forking step that had clear step by step description on how to go about it, this step has no clear description and is very ambiguous. 

The link attached takes the user to a page with a lot of links with no recommendation on which  link to download and how to set up Make nor the differences in installation for the different OS that users may use. Even after downloading a file there was no way to know what to do next in installing the Make command.

**Comment by Chiemezuo** [comment]
Hi @Angamanga 
This project is under the `todo` in [Outreachy 2024: Accessibility review and accessibility improvement of the Ushahidi Platform Client](https://github.com/orgs/ushahidi/projects/18), but the label was changed from "**Outreachy: Accessibility**" to "**Outreachy: Whatsapp**".

Which category does it fall under?

**Comment by tubapervaiz1508** [comment]
Hi @Angamanga 
I wish to work on this issue. I have completed the installation process of the platform. Can you please grant me this work. 

![Image](https://github.com/ushahidi/platform/assets/143891601/1c512de4-9c4b-4458-adbb-15074bd78485)

**Comment by Chiemezuo** [comment]
@Angamanga Here's my comment on the document

# Doc Review
I'll be grouping this comment into the categories of:
- What my setup was already like before Ushahidi.
- What I liked.
- Errors I encountered and how I navigated past them.
- What could be worked on.
- Nice-to-haves (but not entirely necessary).

## My already-existing setup
- Windows 11.
- Docker version 24.0.6 with **WSL 2** enabled.
- Visual Studio Code.
- Git version 2.43.0 With **autocrlf** set to true (This caused some issues I'll discuss)
- GNU Make 4.4.

## What I liked
- Right away, I liked the simplicity of documentation.
- The easy navigation to other segments and the search option.
- The links to external resources.
- The success message to look for and call to action.

## Errors I encountered
I faced a lot of errors which led to more errors.
- `make start` failed a lot initially for no seemingly identifiable reason. I got to realize it was because on my local machine it was saved using the CRLF [EOL](https://itexus.com/glossary/eol-end-of-line/#:~:text=The%20term%20EOL%3A%20End%20of,a%20specific%20action%20or%20operation.) type. I converted appropriately.
- It still failed, because of the EOL types from `artisan` and `docker-php-ext-enable` were of the CRLF EOL type. I had to modify this.
- `make start` failed again because at some point in time, the `platform` service stopped running. I tried restarting the service several times, but it kept stopping. On going through the logs, I realized it was because of some errors in the scripts in the `docker` directory.
- Again, I noticed it was because of the EOL types, so I wrote a script to fix that problem. I did a `cd docker`, then ran `find . -type f -exec sed -i 's/\r$//' {} \;` to fix it, then I ran `cd ..` back to the parent directory, rebuilt and then ran the command again.
- It failed again some minutes after, but on confirming everything was running well, I realized it might have been because of a network fluctuation.
- At some other point in time, the services were not running for reasons I couldn't pinpoint. I retried a number of times and installed **Php** on my computer because it wasn't installed prior that.I did this, and somehow, it worked some time after(although I'm not sure if it was entirely because of that).

## What could be worked on
- The document didn't account for potential errors that people would face. This could be worked on.
- I believe (but have not fully tested) that recommending `autocrlf` be switched to `false` would have resolved most of my errors, because `bash` files and `conf` files are often very senstive to line endings. A potential fix might be running `git config core.autocrlf false` in the folder the project was cloned into.
- An FAQ section might also have been helpful, so that people wouldn't have to ask questions that have already been asked in the past.
- I noticed there wasn't a reference documentation, but I read that it's still being worked on.

## Nice-to-haves
- A video tutorial.
- Modify step 5 [here](https://docs.ushahidi.com/platform-developer-documentation/v/set-up-the-developer-environment-for-ushahidi/set-up-your-developer-environment/setting-up-the-api-with-docker) to enclose ".env.example" and ".env" in backticks (``) for inline code.

## TL;DR
The instructions were clean, but they lacked guidelines on how to fix potential installation errors.

**Comment by shalini1008** [comment]
Subject: Feedback on Ushahidi Platform Setup Documentation

Hello Anna Iosif,

I hope this message finds you well.  I'm Shalini, and I've been going through the documentation for setting up Ushahidi's platform. Overall, it's a valuable resource, but I wanted to share a couple of points where I found the information a bit confusing.

1.Point 3: Installing Make (Parsing Makefile)

Suggestion: Specify the recommended version for Make to download. I propose mentioning either the latest version or a version above 4.0. This would help users ensure compatibility during the installation process.

2.Point 6: Make Start Command

Suggestion: Clarify that the "make start" command will only function if the Makefile is downloaded or integrated into the user's window environment. Providing this information upfront would prevent potential confusion for users during the setup.
I believe these adjustments will enhance the clarity of the documentation and make the setup process smoother for users. Your attention to these details is highly appreciated, and I thank you for your dedication to improving the user experience.

Best regards,
Shalini

**Comment by anjalichaturvedi** [comment]
Hello, I am Anjali Chaturvedi and this is my take on the installation process:

Confusing instructions
1. The installation of the Make command is indicated; however, it would be more understandable if the OS platforms that this instruction relates to are specified. If the user is not familiar with Make or is running an operating system where Make is not easily accessible, they may encounter difficulties.

Errors not mentioned in the documentation
1. Installation or configuration issues with Docker may occur for users. For typical Docker problems, it would be helpful to include troubleshooting advice or connections to relevant resources.
2. When attempting to access the API, users may run into issues if they have other processes open on port 8080. It would be useful to have instructions on how to handle port conflicts.

Getting stuck
If users are unclear about the function of the.env file or experience permission issues when attempting to create or alter files, they may become stopped at the phase of copying text from.env.example into.env.

Additional steps not mentioned
1. Depending on their particular deployment scenario, users might need to configure extra options in the.env file (e.g., database credentials, API keys). It would be beneficial to provide a template or example.env file with placeholders for these settings.
2. No instructions are provided on how to manage or stop the Docker container once it is operating. Providing guidance on the termination of the container would obviate the need for users to look for this information elsewhere.

Suggestions and comments
1. Users who face problems during installation would find it easier to use and less frustrating if there included a troubleshooting area with frequent faults and answers.

**Comment by Tanyadw** [comment]
**Feedback and Suggestions for Improving the Installation Process Documentation**

**Hello , I'm Tanya Dwivedi**
I went through the provided documentation to set up the development environment on my Windows system. While the documentation was helpful overall, I encountered some issues and areas where clarity could be improved. Here are my observations and suggestions for enhancement:

**Confusing instructions:**

- The installation of the Make command is indicated; however, those who are not familiar with Make can encounter issues while installing it. Users might be confused about which version is needed to be downloaded. To address this confusion, I explored alternative methods and found that Make can be installed more simply using Chocolatey.

- The provided link doesn't offer clear guidance on which Make command to download or how to proceed with the installation. Providing specific recommendations and instructions for downloading and installing Make for different operating systems would help clear this ambiguity.

**For Windows Users:**
Open PowerShell in administrative mode.
Run the following commands:
1. Set-ExecutionPolicy Bypass -Scope Process -Force; [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; iex ((New-Object System.Net.WebClient).DownloadString('https://community.chocolatey.org/install.ps1'))
2. choco install make

**For Linux Users:**
On Linux distributions, Make is typically pre-installed. However, if it's not available, you can install it using your distribution's package manager. For example, on Ubuntu or Debian-based systems, you can install Make using apt:
1. sudo apt update
2. sudo apt install make

**For macOS Users:**

Make is often included in the Xcode Command Line Tools package. we can install it by running the following command in the Terminal:
1. xcode-select --install
After this we can verify Make installation by running:  make -v

**Errors not mentioned in the documentation:**
Users may encounter issues with Docker configuration, especially regarding enabling the WSL 2 based engine. Providing instructions and guidance on configuring Docker properly will help users avoid setup problems.

1. In Docker, make sure to tick the use the WSL 2 based engine (windows home can only run the wsl 2 backend) if not tick by default then apply and restart to setup docker within WSL.
2.  Users may encounter issues with Docker configuration, especially regarding enabling the WSL 2 based engine. Providing instructions and guidance on configuring Docker properly will help users avoid setup problems.

**Points of Getting Stuck:**

1. Users might get stuck at the step of enabling Windows Subsystem for Linux (WSL) if they're unable to locate the option within the Control Panel or if they encounter errors during the process. Providing detailed instructions or screenshots for navigating the Control Panel and troubleshooting common WSL setup issues would be helpful.
2. Unclear Function of .env File:
 Users may become stuck at the phase of copying text from .env.example into .env if they are unclear about the function of the .env file. The purpose of the .env file, typically used for configuring environment variables, may not be clear to all users. Providing additional explanations or comments within the documentation regarding the significance of the .env file and how it impacts the application's configuration could help users understand its importance and proceed with confidence.
4. some users might be confused while installing ubuntu ,here I have provided simple steps to install ubuntu.
- Go to the Microsoft store and download the *Ubuntu app*
- Launch the Ubuntu app. After downloading the app and launching it will require you to create a password
These steps are straightforward, but users might encounter difficulties if they're not familiar with the Microsoft Store or Ubuntu app installation process.

**Unmentioned Steps for Making it Work:**

Additional configuration steps, such as setting up environment variables or configuring Docker settings, are essential for the proper functioning of the development environment but are not explicitly mentioned in the documentation. Providing a comprehensive list of prerequisites and additional setup instructions would prevent users from encountering issues due to missing steps.

**Suggestions for Improvement:**

1. Including troubleshooting steps and references for further guidance.
2. Including alternative installation methods or workarounds for common issues to accommodate users with varying setups and environments.
5. Providing video tutorials for an easy setup process could also be beneficial.

Thank you for considering my feedback. I believe implementing these suggestions will significantly improve the clarity and effectiveness of the installation process documentation.
